src/ArrayHelper.php

```diff
@@ -455,22 +455,58 @@ private static function parsePath($path, string $delimiter)
      * ```
      *
      * @param array $array the array to extract value from
-     * @param string $key key name of the array element
+     * @param string|int|float|array $key key name of the array element or associative array at the key path specified
      * @param mixed $default the default value to be returned if the specified key does not exist
      * @return mixed the value of the element if found, default value otherwise
      */
-    public static function remove(array &$array, string $key, $default = null)
+    public static function remove(array &$array, $key, $default = null)
     {
-        if (array_key_exists($key, $array)) {
+        $keys = is_array($key) ? $key : [$key];
```

Comment:
Support for `Closure` can also be added.

Reply:
`Closure` should removed item and return values... Has use cases?